src/hint_processor/hint_processor_utils.zig

```diff
@@ -0,0 +1,127 @@
+const std = @import("std");
+
+const Register = @import("../vm/instructions.zig").Register;
+const ApTracking = @import("../vm/types/programjson.zig").ApTracking;
+const Felt252 = @import("../math/fields/starknet.zig").Felt252;
+const HintReference = @import("hint_processor_def.zig").HintReference;
+const CoreVM = @import("../vm/core.zig");
+const OffsetValue = @import("../vm/types/programjson.zig").OffsetValue;
+const CairoVM = CoreVM.CairoVM;
+const MaybeRelocatable = @import("../vm/memory/relocatable.zig").MaybeRelocatable;
+const Relocatable = @import("../vm/memory/relocatable.zig").Relocatable;
+const Allocator = std.mem.Allocator;
+const HintError = @import("../vm/error.zig").HintError;
+
+///Inserts value into the address of the given ids variable
+pub fn insertValueFromReference(
+    allocator: Allocator,
+    value: MaybeRelocatable,
+    vm: *CairoVM,
+    hint_reference: HintReference,
+    ap_tracking: ApTracking,
+) !void {
+    if (computeAddrFromReference(hint_reference, ap_tracking, vm)) |var_addr| {
+        vm.segments.memory.set(allocator, var_addr, value) catch HintError.Memory;
+    } else return HintError.UnknownIdentifierInternal;
+}
+
+///Returns the Integer value stored in the given ids variable
+/// Returns an internal error, users should map it into a more informative type
+pub fn getIntegerFromReference(
+    vm: *CairoVM,
+    hint_reference: HintReference,
+    ap_tracking: ApTracking,
+) !Felt252 {
+    // if the reference register is none, this means it is an immediate value and we
+    // should return that value.
+    switch (hint_reference.offset1) {
+        .immediate => |int_1| return int_1,
+        else => {},
+    }
+
+    return if (computeAddrFromReference(hint_reference, ap_tracking, vm)) |var_addr| vm.segments.memory.getFelt(var_addr) catch HintError.WrongIdentifierTypeInternal else HintError.UnknownIdentifierInternal;
+}
+
+///Returns the Relocatable value stored in the given ids variable
+pub fn getPtrFromReference(hint_reference: HintReference, ap_tracking: ApTracking, vm: *CairoVM) !Relocatable {
+    const var_addr = if (computeAddrFromReference(hint_reference, ap_tracking, vm)) |addr| addr else return HintError.UnknownIdentifierInternal;
+
+    if (hint_reference.dereference) {
+        return vm.getRelocatable(var_addr) catch HintError.WrongIdentifierTypeInternal;
+    } else return var_addr;
+}
+
+pub fn applyApTrackingCorrection(addr: Relocatable, ref_ap_tracking: ApTracking, hint_ap_tracking: ApTracking) ?Relocatable {
+    if (ref_ap_tracking.group == hint_ap_tracking.group) {
+        return addr.subUint(hint_ap_tracking.offset - ref_ap_tracking.offset) catch unreachable;
```

Comment:
Here why using `unreachable`? If for some reason the `unreachable` is reached then we will have a panic, it is better to detect it with a `try` to bring out the error right?

Reply:
same as rust implementation, we dont have error in stack of calls, that is why we dont handle errors, but anyway if unreacble is reached it will print call stack